lib/dry/struct/class_interface.rb

```diff
@@ -25,22 +25,42 @@ def inherited(klass)
       # and modifies {.schema} accordingly.
       #
       # @param [Symbol] name name of the defined attribute
-      # @param [Dry::Types::Definition] type
+      # @param [Dry::Types::Definition, nil] type or superclass of nested type
       # @return [Dry::Struct]
+      # @yield
+      #   If a block is given, it will be evaluated in the context of
+      #   and new struct class, and set as a nested type for the given
+      #   name
       # @raise [RepeatedAttributeError] when trying to define attribute with the
       #   same name as previously defined one
       #
       # @example
       #   class Language < Dry::Struct
       #     attribute :name, Types::String
+      #     attribute :details, Dry::Struct do
+      #       attribute :type, Types::String
+      #     end
       #   end
       #
       #   Language.schema
-      #     #=> {name: #<Dry::Types::Definition primitive=String options={}>}
+      #     #=> {
+      #           :name=>#<Dry::Types::Definition primitive=String options={} meta={}>,
+      #           :details=>#<Class:0x000055b78c6bd9d0>
+      #         }
       #
-      #   ruby = Language.new(name: 'Ruby')
+      #   ruby = Language.new(name: 'Ruby', details: { type: 'OO' })
       #   ruby.name #=> 'Ruby'
-      def attribute(name, type)
+      #   ruby.details.type #=> 'OO'
+      def attribute(name, type = nil, &block)
+        if block
+          type = build_nested_type(type || ::Dry::Struct, &block)
```

Comment:
Not sure whether type should default to the superclass of the type defining the nested attribute?